### PR TITLE
Pull-Request

### PR DESCRIPTION
**Pull request by Zven94**
In this project, I started building a school library app. In this initial step, I have implemented the classes to represent students and teachers.

Files added:

- Gemfile
- LICENSE
- README
- rubocop.yml
- person.rb
- student.rb
- teacher.rb

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    @parent_permission = parent_permission
+  end
+
+  # define getter for @name
+  attr_accessor :name
+
+  # define setter for @name
+
+  # define getter for @age
+  attr_accessor :age
```
Comment (BregornOriginal) [inline person.rb]: [OPTIONAL] You could define accessor in only one line, like `attr_accessor :name, :age`

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  # define getter for age
+
+  def can_use_services?
+    if (age >= 18) || @parent_permission
+      true
+    else
+      false
+    end
+  end
```
Comment (BregornOriginal) [inline person.rb]: - [x] To follow best practices and clean code. Kindly refactor this method to use the `def of_age?` method that you created as private. In this way you save code and make it easy to read

File: student.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+require_relative 'person'
+
+# creating a Person subclass: student
+class Student < Person
+  # Initializes a Student object with the given classroom
+  def initialize(classroom = 'Unknown', name = 'Unknown', age = 0, parent_permission: true)
```
Comment (BregornOriginal) [inline student.rb]: - [x] Kindly refactor the constructor to follow the needed parameters that the Student will receive. `age, classroom, name = "Unknown", parent_permission = true` The only ones that is defined as default are `name` and `parent_permission`

File: teacher.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+require_relative 'person'
+
+# creating a Person subclass: teacher
+class Teacher < Person
+  def initialize(specialization = 'Unknown', name = 'Unknown', age = 0, parent_permission: true)
```
Comment (BregornOriginal) [inline teacher.rb]: - [x] Kindly refactor the constructor to follow the needed parameters that a Teacher will receive. ``age, specialization, name = "Unknown", parent_permission = true`` The only ones that are defined as default are `name` and `parent_permission`

**Review by BregornOriginal — CHANGES_REQUESTED** [review]
Hi @Zven94  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

giphy (3)

## Highlights

✅ Linters are passing
✅ Good commit history
✅ Excellent work implementing classes 💯 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by usorfaitheloho — APPROVED** [review]
Hi @Zven94  , 
# STATUS: APPROVED! 🥇 💯 

done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**